src/ml-search.service.js

```diff
@@ -1135,11 +1135,55 @@
       var self = this;
       
       // TODO: what other conditions
+      // [GJo] This looks a bit peculiar. If you have start and limit in params, but no options, then they get lost?
       if ( !options && params && params.options) {
         options = params;
         params = null;
       }
```

Comment:
Yeah, but it doesn't describe that the meaning of arguments changes. In Java you would overload the method, and describe them separately.. ;-)